rust/private/lto.bzl

```diff
@@ -94,8 +96,12 @@ def construct_lto_arguments(ctx, toolchain, crate_info):
         list: A list of strings that are valid flags for 'rustc'.
     """
     mode = toolchain._lto.mode
-    format = _determine_lto_object_format(ctx, toolchain, crate_info)
 
+    # The user is handling LTO on their own, don't add any arguments.
+    if mode == "manual":
```

Comment:
We can definitely iterate! Are you referring to the `elif format == "only_bitcode"` [condition](https://github.com/bazelbuild/rules_rust/blob/main/rust/private/lto.bzl#L109-L112)?

Reply:
I was referring to these which are not working today correctly. If we choose this approach we should fix the below conditions according to the new behavior.
```
    if format in ["unspecified", "object_and_bitcode"]:
        # Embedding LLVM bitcode in object files is `rustc's` default.
        args.extend([])
    elif format in ["off", "only_object"]:
        args.extend(["embed-bitcode=no"])
```

Reply:
Sweet, more than happy to change these if you can describe how they're broken? If you can share an error message you're getting from a build that would be great!

I ask because I've tested all variants of the LTO setting on a relatively large project that links a few large C libraries (e.g. `librdkafka`, `rocksdb`, `openssl`) and all of the builds succeed?

Reply:
I mean that they logically wrong. Based on what you are proposing it should look something like this in my opinion:

```
    if format == "object_and_bitcode":
        # Embedding LLVM bitcode in object files is `rustc's` default.
        args.extend([])
    elif mode in ["unspecified", "off"] or format == "only_object":
        args.extend(["embed-bitcode=no"])
```

Reply:
I see, thanks for pointing this out!

As-is the existing implementation is logically the same as you've written here. I think where the disconnect comes from is how I still reference "unspecified" and "off" w.r.t. `format`. In a previous iteration of the original PR they were formats but I realized that was confusing, so I changed [`_determine_lto_object_format` to return one of "only_object", "only_bitcode", or "object_and_bitcode"](https://github.com/bazelbuild/rules_rust/blob/main/rust/private/lto.bzl#L53). "unspecified" and "off" already [map to "only_object"](https://github.com/bazelbuild/rules_rust/blob/main/rust/private/lto.bzl#L63-L64) but making this clearer is a good call!